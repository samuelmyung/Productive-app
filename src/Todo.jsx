import React from "react";

/** Simple presentation component for a todo.
 *
 * Props:
 * - todo: like { id, title, description, priority, completed }
 * - update(): fn to call to update a todo
 *
 * { EditableTodo, TopTodo } -> Todo
 **/

function Todo({ id, title, description, priority, completed = false, update }) {
  const style = {
    textDecoration: completed ? 'line-through' : 'none'
  }

  function toggleCompleted() {
    update({
      id,
      title,
      description,
      priority,
      completed: !completed
    });
  }

  return (
    <div className="Todo" key={id}>
      <div onClick={toggleCompleted}><b style={style}>{title}</b> <small>(priority: {priority})</small></div>
      <div><small>{description}</small></div>
    </div>
  );
}

export default Todo;
