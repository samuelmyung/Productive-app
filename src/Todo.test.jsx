import { it, expect } from "vitest";

import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

it('renders without crashing', function () {
    render(
        <Todo
            id={1}
            title={'test title'}
            description={'test description'}
            priority={1}
        />
    );
});

it('matches snapshot with initial todos', function () {
    const {container} = render(
    <Todo
        id={1}
        title={'test title'}
        description={'test description'}
        priority={1}
    />
    );
    expect(container).toMatchSnapshot();
});