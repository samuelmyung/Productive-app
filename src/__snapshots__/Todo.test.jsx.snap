// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot with initial todos 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        test title
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        test description
      </small>
    </div>
  </div>
</div>
`;
